modules/shortcodes/soundcloud.php

```diff
@@ -109,7 +110,17 @@ function soundcloud_shortcode( $atts, $content = null ) {
 		$options['visual'] = false;
 	}
 
-	// Build our list of Souncloud parameters.
+	if (
+		class_exists( 'Jetpack_AMP_Support' )
+		&& Jetpack_AMP_Support::is_amp_request()
+		&& isset( $options['url'] )
+		&& 'api.soundcloud.com' !== wp_parse_url( $options['url'], PHP_URL_HOST )
+	) {
+		// Defer to oEmbed if an oEmbeddable URL is provided.
+		return $wp_embed->shortcode( $options, $options['url'] );
+	}
```

Comment:
This [is copied](https://github.com/ampproject/amp-wp/blob/develop/includes/embeds/class-amp-soundcloud-embed.php#L138-L142) from the AMP plugin.

Calling `$wp_embed->shortcode( ... )` eventually triggers an oembed filter in the AMP plugin, [filter_embed_oembed_html()](https://github.com/ampproject/amp-wp/blob/develop/includes/embeds/class-amp-soundcloud-embed.php#L67). 

This is part of the AMP plugin, and can probably stay there. It's also used for oEmbed conversion, like if you simply entered a SoundCloud URL a post.